core/src/main/java/org/wso2/msf4j/interceptor/MSF4JRequestInterceptor.java

```diff
@@ -0,0 +1,71 @@
+/*
+*  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package org.wso2.msf4j.interceptor;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.msf4j.Request;
+import org.wso2.msf4j.Response;
+
+/**
+ * Interface that needs to be implemented to intercept request method calls
+ * <p>
+ * Priority of request interceptors will be
+ * Global request interceptors -> Class level annotated request interceptors
+ * -> Method level annotated interceptors -> Resource method execution
+ * -> sub-resource method class level annotated interceptors -> sub-resource method method level annotated interceptors
+ * -> sub-resource execution
+ * <p>
+ * Upon returning false from any interceptor will cause the above flow to break and send the response immediately
+ * <p>
+ * Upon exception in interceptors "onRequestInterceptionError" default method in the interface will be called.
+ * Override this method for custom behaviour
+ */
+@FunctionalInterface
+public interface MSF4JRequestInterceptor {
```

Comment:
Removed the MSF4J prefix as requested. The interface names would be RequestInterceptor and ResponseInterceptor. When the developer is using MSF4J he will not get an opportunity to use both classes in a single class in a typical scenario.